### PR TITLE
[JENKINS-71715] Update minimum supported Jenkins version to 2.440 to be able to use new `ConsoleUrlProvider` extension point to redirect to the console

### PR DESCRIPTION
**Pull request by jgreffe**
See [JENKINS-71715](https://issues.jenkins.io/browse/JENKINS-71715) and https://github.com/jenkinsci/jenkins/pull/8321.

### Testing done

Tested in combination with https://github.com/jenkinsci/workflow-job-plugin/pull/373 and https://github.com/jenkinsci/pipeline-graph-view-plugin/pull/289, see this video:

https://github.com/jenkinsci/pipeline-input-step-plugin/assets/1068968/de3788fb-c41f-48ba-9420-561e82e2a354

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java [hunk 1]
```diff
@@ -199,14 +200,15 @@ public String getDisplayName() {
...
+        Run<?, ?> run = getRun();
         if (request.getParameter("proceed")!=null) {
             doProceed(request);
         } else {
             doAbort();
         }
 
         // go back to the Run console page
-        return HttpResponses.redirectTo("../../console");
+        return HttpResponses.redirectTo(ConsoleURLProvider.get().getConsoleURL(run));
```
Comment (dwnusbaum) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java:211]: Hmm, thinking about this, I wonder if we should encourage plugins like this to use `ConsoleURLProvider` or `DisplayURLProvider`. `ConsoleURLProvider` is only really needed for links in Jenkins core itself. Plugins could use `DisplayURLProvider.getConsoleURL` right away if we split the `display-url-api` PR into two parts, one just for `DisplayURLProvider.getConsoleURL` and one to update the core dependency and implement `ConsoleURLProvider`.

Comment (jgreffe) [reply src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java:211]: If `display-url-api` is not installed, the `DisplayURLProvider` is not available from the current plugin, where `ConsoleURLProvider` is.

Comment (dwnusbaum) [reply src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java:211]: Yes, but I think it should not be a big deal to add `display-url-api` as a compile-scope dependency to plugins that produce console links.

Comment (dwnusbaum) [reply src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java]: Outdated based on upstream changes, see https://github.com/jenkinsci/pipeline-input-step-plugin/pull/145/commits/fd34d23a733c82aef6f80f52d511d945d0ca6f9d.

File: pom.xml [hunk 2]
```diff
@@ -39,6 +39,7 @@
     <properties>
         <changelist>999999-SNAPSHOT</changelist>
         <jenkins.version>2.361.4</jenkins.version>
+        <jenkins.version>2.418-rc34068.6491cb_b_a_fd22</jenkins.version> <!-- TODO: https://github.com/jenkinsci/jenkins/pull/8321 -->
```
Comment (dwnusbaum) [inline pom.xml:42]:
```suggestion
        <jenkins.version>2.420-rc34121.c34df9dd8347</jenkins.version> <!-- TODO: https://github.com/jenkinsci/jenkins/pull/8321 -->
```

File: pom.xml [hunk 3]
```diff
@@ -38,7 +38,7 @@
     </pluginRepositories>
     <properties>
         <changelist>999999-SNAPSHOT</changelist>
-        <jenkins.version>2.401.3</jenkins.version>
+        <jenkins.version>2.420-rc34121.c34df9dd8347</jenkins.version> <!-- TODO: https://github.com/jenkinsci/jenkins/pull/8321 -->
```
Comment (timja) [inline pom.xml:41]:
```suggestion
        <jenkins.version>2.440</jenkins.version>
```

**Comment by timja** [comment]
ready to go @dwnusbaum?

**Review by jglick — APPROVED** [review]

**Comment by jglick** [comment]
> Needs to be retested.

Still true?

**Comment by dwnusbaum** [comment]
> Still true?

I added a video to the PR description showing the new behavior.

**Review by dwnusbaum — APPROVED** [review]

**Pull request merged**